File: tests/fixtures/edgecase_repo/rust/src/module5.rs
use crate::module4::Service4;

pub struct Base5 {
    pub name: String,
}

pub struct Service5 {
    pub base: Base5,
}

pub trait Handler5 {
    fn handle(&self, payload: &str) -> bool;
}

impl Handler5 for Service5 {
    fn handle(&self, payload: &str) -> bool {
        self.validate(payload);
        true
    }
}

impl Service5 {
    fn validate(&self, payload: &str) {}
}

pub fn helper_5(value: &str) -> &str {
    value
}

File: tests/fixtures/edgecase_repo/javascript/src/module5.js
const { Service4 } = require("./module4");

class Base5 {
  constructor(name) {
    this.name = name;
  }
}

class Service5 extends Base5 {
  handle(payload) {
    this.validate(payload);
    return new Service4("x").handle(payload);
  }

  validate(payload) {
    return payload !== null;
  }
}

function helper5(value) {
  return value;
}

module.exports = { Service5 };
